File: replace_colors.js
const fs = require('fs');
const path = require('path');

const colors = {
    '#1243B2': '#10367D',
    '#A9CDE5': '#74B4DA',
    '#F8FAFC': '#EBEBEB'
};

function replaceInFile(filepath) {
    try {
        const content = fs.readFileSync(filepath, 'utf8');
        let newContent = content;
        for (const [oldColor, newColor] of Object.entries(colors)) {
            newContent = newContent.split(oldColor).join(newColor);
        }
        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Updated: ${filepath}`);
        }
    } catch (err) {
        // Silently skip binary files or permission issues
    }
}

function walk(dir) {
    try {
        const entries = fs.readdirSync(dir, { withFileTypes: true });
        for (const entry of entries) {
            const filepath = path.join(dir, entry.name);
            if (entry.isDirectory()) {
                if (!['node_modules', '.next', '.git'].includes(entry.name)) {
                    walk(filepath);
                }
            } else if (/\.(tsx|ts|css|js|mjs)$/.test(entry.name)) {
                replaceInFile(filepath);
            }
        }
    } catch (err) {
        // console.error(`Walk error in ${dir}: ${err.message}`);
    }
}

walk('.');
console.log('Migration Completed.');

File: replace_colors.py
import os

colors = {
    '#1243B2': '#10367D',
    '#A9CDE5': '#74B4DA',
    '#F8FAFC': '#EBEBEB'
}

def replace_in_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        new_content = content
        for old, new in colors.items():
            new_content = new_content.replace(old, new)
        
        if new_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated: {filepath}")
    except Exception as e:
        pass

for root, dirs, files in os.walk('.'):
    if any(d in root for d in ['node_modules', '.next', '.git']):
        continue
    
    for file in files:
        if file.endswith(('.tsx', '.ts', '.css', '.js', '.mjs')):
            replace_in_file(os.path.join(root, file))
